core/src/main/scala/org/apache/spark/sql/sources/RuleUtils.scala

```diff
@@ -38,16 +39,19 @@ import org.apache.spark.sql.sources.Entity.{INDEX, INDEX_RELATION, TABLE}
 
 object RuleUtils extends PredicateHelper {
 
+  private def getIndex(catalog: SnappyStoreHiveCatalog, name: String) = {
+    val relation = catalog.lookupRelation(catalog.newQualifiedTableName(name))
+    relation match {
+      case LogicalRelation(i: IndexColumnFormatRelation, _, _) => Some(relation)
+      case _ => None
+    }
+  }
+
   def fetchIndexes(snappySession: SnappySession,
       table: LogicalPlan): Seq[(LogicalPlan, Seq[LogicalPlan])] = table.collect {
     case l@LogicalRelation(p: ParentRelation, _, _) =>
       val catalog = snappySession.sessionCatalog
-      (l.asInstanceOf[LogicalPlan], p.getDependents(catalog).map(idx =>
-        catalog.lookupRelation(catalog.newQualifiedTableName(idx))))
-    case s@SubqueryAlias(alias, child@LogicalRelation(p: ParentRelation, _, _)) =>
```

Comment:
No. It wasn't earlier when rule was getting applied at the analyzer level. Off late this rule is shifted to optimizer level and EliminateSubqueryAlias rule gets applied first and therefore now its not required to handle. Nothing to do with inheritence.